router/io-spec-non-proven-lemmas.gobra

```diff
@@ -166,6 +166,68 @@ pure func (p* scionPacketProcessor) GetIsXoverSpec(ub []byte) bool {
 		p.path.GetIsXoverSpec(ubPath)
 }
 
+ghost
+opaque
+requires acc(&p.d, R55) && acc(p.d.Mem(), _)
+requires acc(&p.ingressID, R55)
+requires len(pkt.CurrSeg.Future) > 0
+decreases
+pure func (p *scionPacketProcessor) NoBouncingPkt(pkt io.IO_pkt2) bool {
+	return let currseg := pkt.CurrSeg in
+		let OptEgressID := (currseg.ConsDir ? currseg.Future[0].EgIF2 : currseg.Future[0].InIF2) in
```

Comment:
this operation seems to be performed in a few places, including `AbsEgressInterfaceConstraint`. Couldn't we abstract it in a function? (a suggestive name could be `EgressIdInDir()`)

Reply:
I agree there are many places that look similar. However `(currseg.ConsDir ? currseg.Future[0].EgIF2 : currseg.Future[0].InIF2)`is actually only used twice. I don't know if it's worth introducing a function for that?

Reply:
the advantage of having this as a function is that it becomes usable in the future already if at some point we need to use these expressions. Also, it makes it easier to read this function and grasp what it means. I would change it if the resulting function does not require too much annotation overhead